File: cheese/Optml.Html.Tests/Testing.Fable.js
import { Mocha_runTests, Test_ptestCase, Test_testCase, Test_testList } from "../.fable/Fable.Mocha.2.11.0/Mocha.fs.js";
import { Attribute } from "../.fable/fable-library.3.2.9/Types.js";
import { class_type } from "../.fable/fable-library.3.2.9/Reflection.js";

export const testList = (name) => ((tests) => Test_testList(name, tests));

export const testCase = (name) => ((body) => Test_testCase(name, body));

export const ptestCase = (name) => ((body) => Test_ptestCase(name, body));

export class TestsAttribute extends Attribute {
    constructor() {
        super();
    }
}

export function TestsAttribute$reflection() {
    return class_type("Optml.Html.Testing.TestsAttribute", void 0, TestsAttribute, class_type("System.Attribute"));
}

export function TestsAttribute_$ctor() {
    return new TestsAttribute();
}

export function run(_arg1, tests) {
    return Mocha_runTests(tests);
}

File: Optml.Html.Tests/Testing.Fable.fs
module Optml.Html.Testing

let testList = Fable.Mocha.Test.testList
let testCase = Fable.Mocha.Test.testCase
let ptestCase = Fable.Mocha.Test.ptestCase

/// Dummy Tests attribute type (to replace the Expecto one which we use when running on .NET)
type TestsAttribute() = inherit System.Attribute()

module Expect =
  // Inline because otherwise:
  // FABLE: Cannot get type info of generic parameter a$. Fable erases generics at runtime, try inlining the functions so generics can be resolved at compile time.
  let inline equal expected actual = Fable.Mocha.Expect.equal actual expected ""
  let inline isTrue actual = Fable.Mocha.Expect.isTrue actual ""
  let inline isFalse actual = Fable.Mocha.Expect.isFalse actual ""
  let inline hasLength expected actual = Fable.Mocha.Expect.equal (Seq.length actual) expected ""
  let inline stringStarts (prefix : string) (actual : string) = Fable.Mocha.Expect.isTrue (actual.StartsWith prefix) ""
  let inline wantSome actual =
    Fable.Mocha.Expect.isSome actual ""
    actual.Value
  let inline isNone actual = Fable.Mocha.Expect.isNone actual ""

let run _ tests = Fable.Mocha.Mocha.runTests tests
